src/cargo/core/compiler/build_context/target_info.rs

```diff
@@ -277,6 +292,22 @@ impl TargetInfo {
                 res
             };
 
+            let check_cfg = if gctx.cli_unstable().check_target_cfgs {
+                let mut check_cfg = CheckCfg::default();
+                check_cfg.exhaustive = true;
```

Comment:
API-wise, this feels dirty that we are having to tell it that the default state before parsing is `exhaustive = true`

Reply:
We only know if it's exhaustive by the absence of `any()=any()` and `any()`, which we can't know inside `parse_print_check_cfg_line` since it doesn't know if we have finished processing all the lines.

What about having associated function `CheckCfg::parse_print_check_cfg_lines` (note the `s` at the end), which would return a `CheckCfg` (no Default impl anymore), this function would do the business of parsing all the lines and setting the `exhaustive` flag appropriately.